### ISSUE TITLE
RAMP: Docker setup

### ISSUE THREAD

**Issue by rth** [paris-saclay-cds/ramp-board#119]
I'm planning to try to deploy RAMP backend for teaching and was considering to put it in containers so I can isolate it from the server where it is going to run.

I was thinking to make 3 containers database / engine / dashboard, then link them in a [`docker-compose.yml`](https://docs.docker.com/compose/overview). 

Did you discuss it before? Maybe you already have some thoughts on how this could be better achieved? Thanks!

**Comment by rth** [comment]
Probably the whole thing while I'm at it (that's why I was thinking about 3 sepearate containers), not so much for security but to make sure I can easily re-deploy it easily it on another server if needed.

**Comment by kegl** [comment]
I would first make a single docker repo that contains the server and the engine and check. Then factorize if needed.

**Comment by rth** [comment]
Sounds good. Thanks, will try that.

**Comment by rth** [comment]
A status update on this is that I initially tried to run everything in a single docker instance. After soem work, I realized that a number of installations steps require a running postgres service which is difficult to do in the Dockerfile. So I then tried to run postgres on the host system and connect to it from within the container. Except that this also was non-trivial (some virtual network interface configuration was necessary I think) so I abandoned that as well as I didn't want to spend much time on it.

In the end, I'm running RAMP on regular Ubuntu server which works.
